ui/blocks/AffiliateCard.tsx

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react'
+import { Affiliate, findAffiliateByCode } from '~lib/api/affiliates'
+import { Card } from '~ui/core'
+
+interface Props {
+  code: string
+}
+
+export const AffiliateCard: React.FC<Props> = ({ code }) => {
+  const [affiliate, setAffiliate] = React.useState<Affiliate>(null)
+
+  React.useEffect(() => {
+    if (code === undefined) return
+    findAffiliateByCode(code)
+      .then((affiliate) => setAffiliate(affiliate))
+      .catch(() => {
+        setAffiliate(null)
+      })
+  }, [code])
+
+  return affiliate ? (
```

Comment:
I think that both `logoLink` and `logoUrl` should be checked before rendering anything
```suggestion
  return affiliate?.logoLink && affiliate?.logoUrl ? (
```
In case `logoLink` is optional, then an extra block should be added for rendering the logo without the link wrapper.

Reply:
See above, I do not think we need this complexity. If affiliate?.logoLink is null, the result is as expected.

Reply:
Ah I think I see your point.

Reply:
I suggest this change:

```
  return affiliate.logoUrl ? (
    <Card style={{ padding: '1em' }}>
      <a href={affiliate.logoLink} target="_blank" rel="noreferrer noopener">
        <img
          alt="affilitate logo"
          style={{ maxWidth: '100%' }}
          src={affiliate.logoUrl}
        ></img>
      </a>
    </Card>
  ) : (
    <></>
  )
```

Reply:
@salimhb see my latest commit.